src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -827,34 +827,9 @@ internal static int GetHashCodeOrdinalIgnoreCase(ReadOnlySpan<char> value)
         // restructure the comparison so that for odd-length spans, we simulate the null terminator and include
         // it in the hash computation exactly as does str.GetNonRandomizedHashCode().
 
-        internal unsafe int GetNonRandomizedHashCode()
+        internal int GetNonRandomizedHashCode()
         {
-            fixed (char* src = &_firstChar)
-            {
-                Debug.Assert(src[Length] == '\0', "src[Length] == '\\0'");
-                Debug.Assert(((int)src) % 4 == 0, "Managed string should start at 4 bytes boundary");
-
-                uint hash1 = (5381 << 16) + 5381;
-                uint hash2 = hash1;
-
-                uint* ptr = (uint*)src;
-                int length = Length;
-
-                while (length > 2)
-                {
-                    length -= 4;
-                    hash1 = (BitOperations.RotateLeft(hash1, 5) + hash1) ^ ptr[0];
-                    hash2 = (BitOperations.RotateLeft(hash2, 5) + hash2) ^ ptr[1];
-                    ptr += 2;
-                }
-
-                if (length > 0)
-                {
-                    hash2 = (BitOperations.RotateLeft(hash2, 5) + hash2) ^ ptr[0];
-                }
-
-                return (int)(hash1 + (hash2 * 1566083941));
-            }
+            return GetNonRandomizedHashCode(new ReadOnlySpan<char>(ref _firstChar, _stringLength));
```

Comment:
Does this negatively impact perf?

The implementation as it existed was benefiting from strings being null terminated. The span-based overload can't rely on that and pays a tad for it.

Reply:
At the same time the Span one special cased 1,2,3 length, it was an improvement last time I ran benchmark, I'll do again